include/extractor/guidance/turn_lane_types.hpp

```diff
@@ -95,6 +96,11 @@ struct TurnLaneDescription_hash
     }
 };
 
+typedef std::unordered_map<guidance::TurnLaneDescription,
+                           LaneDescriptionID,
+                           guidance::TurnLaneDescription_hash>
```

Comment:
I just had a look at the hash struct since I'm seeing this for the first time.
There is `hash_range` providing the effect you need there:
http://www.boost.org/doc/libs/1_55_0/doc/html/hash/reference.html#idp53740664-bb